scheduler/scheduler.go

```diff
@@ -179,6 +193,18 @@ func (fq *fifoQueue) wait(ctx context.Context) (func(), error) {
 			// does not exist yet in Go stdlib as of Go1.18.4
 			return nil, err
 		}
+	} else {
+		// semaphore will not fail on Acquire when context timed out, nor when it's canceled.
+		// see also: https://github.com/golang/go/issues/63615
+		// The behavior can change as in the proposed change, so the code path on error would apply.
+		// If the proposed change was merge this code path below can likely be cleaned up.
+		select {
+		case <-c.Done():
+			// We timed out in Acquire, so it never increased semaphore by one
+			cnt.Add(-1)
+			return nil, ErrQueueTimeout
+		default:
+		}
```

Comment:
I think we don't need this part as we checked that context is not expired before Acquire.
If we have it then we should use ctx.Err() instead of select to make it all consistent.

Reply:
I was thinking about the case when request times out when we block in the Acquire() call.
But I think we tested it and observed that this would return error, so yes I will cleanup

Reply:
fixed